common/display/displayqueue.cpp

```diff
@@ -437,7 +437,8 @@ void DisplayQueue::GetCachedLayers(const std::vector<OverlayLayer>& layers,
         reset_composition_regions = true;
 
       last_plane.SetOverlayLayer(layer);
-      if (layer->HasLayerContentChanged()) {
+      if (layer->HasLayerContentChanged() ||
+          layer->GetSurfaceDamage().empty()) {
```

Comment:
@Shao-Feng 
Can't we change HasLayerContentChanged() function to add this WA?
It will be better I think

Reply:
I hope we don't need to merge this one. Just commit here for tracing the bug. I hope to find out the root cause why screen is flicking  after the unchanged frame-commit is ignored. The logic should be correct. No need to commit for the un-changed frame before sleeping. Unless the buffer of layer is changed in turning to sleep. but SF did not inform us

Reply:
Is it possible the issue is in HasLayerContentChanged()? since SF use cache call in new AOSP, after resume from S3, layer cached is cleared, but hwc still think the layer is not changed? just a guess.

Reply:
The flicking happen before powermode is set as kOff. not in resuming.